### PR TITLE
Ana/4760 add substrate unstake button

### LINKED ISSUE #4760: Substrate Unstake button on Portfolio
add unstake button on portfolio:

- opens action modal with no validator selection option

- only for substrate chains

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #4760 

**Description:**

Easy-peasy

image

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f440de17d14cf60f0bbb210--lunieio.netlify.app

**Comment by Bitcoinera** [comment]
networkFees query in ActionModal not working when going through the Portfolio UnstakeModal... so strange

**Comment by jbibla** [comment]
i think the button should go in the balance row

**Comment by Bitcoinera** [comment]
> i think the button should go in the balance row

Yes, it is also there, it follows the Send button logic, just forgot to upload a screenshot with that view:

image

**Comment by jbibla** [comment]
perfect

File: app/src/ActionModal/components/UndelegationModal.vue [hunk 1]
```diff
@@ -52,7 +52,7 @@
       </div>
     </TmFormGroup>
     <TmFormGroup
-      v-if="session.addressRole !== `stash`"
+      v-if="!isGlobalUnstake && session.addressRole !== `stash`"
```
Comment (faboweb) [inline app/src/ActionModal/components/UndelegationModal.vue:55]: why not `!sourceValidator`? saves you a whole prop

File: app/src/components/common/TmBalance.vue [hunk 2]
```diff
@@ -209,6 +225,11 @@ export default {
       return this.networks.find((network) => network.id === this.network)
         .testnet
     },
+    allDelegationsStake() {
```
Comment (faboweb) [inline app/src/components/common/TmBalance.vue:228]: remove? it is not even used

File: app/src/ActionModal/components/UndelegationModal.vue (hunk 1 above)
Comment (Bitcoinera) [reply app/src/ActionModal/components/UndelegationModal.vue:55]: true

**Comment by faboweb** [comment]
The button doesn't show:

image

**Comment by Bitcoinera** [comment]
> The button doesn't show:

Ah, because of the BalanceRow PR... I see

**Comment by Bitcoinera** [comment]
> The button doesn't show:

Fixed :v:

File: app/src/components/staking/DelegationsOverview.vue [hunk 3]
```diff
@@ -52,6 +57,14 @@ export default {
   computed: {
     ...mapState([`session`]),
     ...mapGetters([`address`, `network`, `networks`]),
+    stakedBalance() {
+      const liquidBalance =
+        Number(this.balances[0].total) - Number(this.balances[0].available)
```
Comment (faboweb) [inline app/src/components/staking/DelegationsOverview.vue:62]: this needs to be filtered by the stakingDenom not just taking the first balance

Comment (Bitcoinera) [reply app/src/components/staking/DelegationsOverview.vue]: true

**Pull request merged**